Mathlib/Combinatorics/SimpleGraph/Connectivity/Represents.lean

```diff
@@ -70,4 +74,20 @@ lemma ncard_sdiff_of_not_mem (hrep : Represents s C) (h : c ∉ C) :
     (c.supp \ s).ncard = c.supp.ncard := by
   rw [(disjoint_supp_of_not_mem hrep h).sdiff_eq_right]
 
-end SimpleGraph.ConnectedComponent.Represents
+end ConnectedComponent.Represents
+
+/-- The odd components are the connected components of odd cardinality. This definition excludes
+infinite components. -/
+def oddComponents : Set G.ConnectedComponent := {c : G.ConnectedComponent | Odd c.supp.ncard}
```

Comment:
Could you make a separate PR that
* defines `oddComponents`
* uses it in your existing lemmas in place of `{c : G.ConnectedComponent | Odd c.supp.ncard}`

?

Reply:
Done:) #22125